src/main/scala/Covid19/Sources/WorldSource.scala

```diff
@@ -28,30 +26,31 @@ final class Jhu(implicit backend: SttpBackend[Identity, Nothing, NothingT], impl
   }
 
   private def getConfirmedByCountry(countryNames: Seq[String]): IO[Confirmed] =
-    getSummaryByCountryByCategory[Confirmed](countryNames).map(Confirmed)
+    getSummaryByCountryByCategory[Confirmed](countryNames).map(Confirmed("", _))
 
   private def getDeadByCountry(countryNames: Seq[String]): IO[Dead] =
-    getSummaryByCountryByCategory[Dead](countryNames).map(Dead)
+    getSummaryByCountryByCategory[Dead](countryNames).map(Dead("", _))
 
   private def getRecoveredByCountry(countryNames: Seq[String]): IO[Recovered] =
-    getSummaryByCountryByCategory[Recovered](countryNames).map(Recovered)
+    getSummaryByCountryByCategory[Recovered](countryNames).map(Recovered("", _))
 
   private def getSummaryByCountryByCategory[C <: InfectedCategory: CategoryName](countryNames: Seq[String]): IO[Int] = {
     val requestIo = IO.pure(basicRequest.get(uri"${baseUrl}time_series_covid19_${CategoryName[C].name}_global.csv"))
 
     requestIo
       .map(request => request.send().body)
       .map {
-        response => response.fold(_ => 0, extract(_, countryNames))
+        response => response.fold(_ => 0, extractByCountry(_, countryNames))
       }
   }
 
-  private def extract(data: String, countryNames: Seq[String]): Int = {
-    data.
-      split("\n").toList.
-      map(_.split(",").toList).
-      filter(l => countryNames.contains(l(1))).
-      map(_.last.trim.toInt).sum
+  private def extractByCountry(data: String, countryNames: Seq[String]): Int = {
+    data
+      .split("\n")
+      .map(_.split(","))
+      .filter(l => countryNames.contains(l(1)))
```

Comment:
why do you need this?

Reply:
To filter data by required country. Actually, this functionality can be implemented via `getInfected` method.